cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -1365,3 +1258,599 @@ static hb_mc_idx_t hb_mc_get_tile_id (hb_mc_coordinate_t origin, hb_mc_dimension
+	size_t sz;
+	error = hb_mc_eva_to_npa(cfg, map, coord, &org_x_eva, &org_x_npa, &sz);
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grp_org_x npa from eva.\n", __func__);
+		return error;
+	}
+
+	error = hb_mc_eva_to_npa(cfg, map, coord, &org_y_eva, &org_y_npa, &sz);
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grp_org_y npa from eva.\n", __func__);
+		return error;
+	}
+
+
+	error = hb_mc_manycore_write32(mc, &org_x_npa, hb_mc_coordinate_get_x(*origin));
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err(	"%s: failed to write __bsg_grp_org_x to tile (%d,%d).\n",
+				__func__,
+				hb_mc_coordinate_get_x(*coord),
+				hb_mc_coordinate_get_y(*coord)); 
+		return error;
+	}
+	bsg_pr_dbg(	"%s: Setting tile (%d,%d) __bsg_grp_org_x (epa 0x%08" PRIx32 ") to %d.\n",
+			__func__,
+			hb_mc_coordinate_get_x(*coord), hb_mc_coordinate_get_y(*coord),
+			org_x_npa.epa, hb_mc_coordinate_get_x(*origin));
+
+
+	error = hb_mc_manycore_write32(mc, &org_y_npa, hb_mc_coordinate_get_y(*origin));
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err(	"%s: failed to write __bsg_grp_org_y to tile (%d,%d).\n",
+				__func__,
+				hb_mc_coordinate_get_x(*coord),
+				hb_mc_coordinate_get_y(*coord)); 
+		return error;
+	}
+	bsg_pr_dbg(	"%s: Setting tile (%d,%d) __bsg_grp_org_y (epa 0x%08" PRIx32 ") to %d.\n",
+			__func__,
+			hb_mc_coordinate_get_x(*coord),
+			hb_mc_coordinate_get_y(*coord),
+			org_y_npa.epa,
+			hb_mc_coordinate_get_y(*origin));
+
+	return HB_MC_SUCCESS;
+}
+
+
+
+
+/*!
+ * Sets a Vanilla Core Endpoint's tile group's coordinate symbols __bsg_x/y.
+ * Behavior is undefined if #mc is not initialized with hb_mc_manycore_init().
+ * @param[in] mc         A manycore instance initialized with hb_mc_manycore_init().
+ * @param[in] map        Eva to npa mapping. 
+ * @param[in] bin        Binary elf file. 
+ * @param[in] bin_size   Size of binary file. 
+ * @param[in] coord      Tile coordinates to set the coordinates of.
+ * @param[in] coord_val  The coordinates to set the tile.
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure.
+ */
+int hb_mc_tile_set_coord_symbols (	hb_mc_manycore_t *mc,
+					hb_mc_eva_map_t *map,
+					unsigned char* bin,
+					size_t bin_size,
+					const hb_mc_coordinate_t *coord,
+					const hb_mc_coordinate_t *coord_val){
+
+	int error;
+	const hb_mc_config_t *cfg = hb_mc_manycore_get_config (mc); 
+
+	hb_mc_eva_t bsg_x_eva, bsg_y_eva;
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_x", &bsg_x_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_x eva.\n", __func__);
+		return HB_MC_NOTFOUND;
+	}
+
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_y", &bsg_y_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_y eva.\n", __func__);
+		return HB_MC_NOTFOUND;
+	}
+
+
+	hb_mc_npa_t bsg_x_npa, bsg_y_npa;
+	size_t sz;
+	error = hb_mc_eva_to_npa(cfg, map, coord, &bsg_x_eva, &bsg_x_npa, &sz);
```

Comment:
There's a function declared in `bsg_manycore_eva.h` called `bsg_manycore_eva_write()` that does Eva translation and writing all at once. You don't need to change your code but that's something to know for future reference.

Reply:
I would argue, that where we can, we should use the appropriate functions.

Reply:
OK. @bornaehsani please address this.

Reply:
Oh you're right! Thanks for pointing it out, I'll take care of them!